### DISCUSSION TITLE
OIDC mit Azure AD Auth ohne Keycloak?

### DISCUSSION THREAD

**Discussion by zlex650** [urlaubsverwaltung/zeiterfassung#348] [Q&A] [answered]
Hallo zusammen, 

gibt es denn inzwischen eine Lösung und vor allem eine Anleitung die Zeiterfassung gegen Azure AD ohne Keycloak als Zwischensystem für die Authentifizierung zu konfigurieren und zu starten?

Gruß Alex

**Comment by grafjo** [comment]
Hi @zlex650,

nein gibt es aktuell noch nicht.

Viele Grüße

**Reply by zlex650** [reply]
Hi @grafjo,

schade. aber gut. Aber dann habe ich noch eine Frage bzgl. Session Timeout. Wo kann man das denn anpassen? Alle änderungen bisher in Keycloak brachten keinen Erfolg. ist der in der Zeiterfassungsapp codiert?

Gruß Alex

**Reply by grafjo** [reply]
also es gibt sowohl in der zeiterfassung als auch in keycloak einen session timeout. wenn du den browser (tab) offen lässt und dich nicht neu anmelden willst, kannst du den timeout in der zeiterfassung erhöhen. wenn du deine zugangsdaten in keycloak nicht neu eingeben willst, könntest du dort den session timeout etwas höher machen und "remember me" anbieten (und dann auch verwenden).
in der zeiterfassung kannst du sen session timeout via `server.servlet.xxx` steuern, da musst du bei spring boot in der ref doku nachlesen, welches die konkrete property ist.

**Reply by zlex650** [reply]
Super Danke für die Info, die Property SERVER_SERVLET_SESSION_TIMEOUT brachte so wie es aussieht die Lösung!👍

**Comment by hesne** [comment]
gibt es hierzu Neuigkeiten? Geht dies zwischenzeitlich?

**Reply by grafjo** [reply]
nein, das geht bisher noch nicht

**Comment by zlex650** [comment]
Habe das nicht weiter verfolgt und auch nichts aktuelleres gehört. Sicherlich wäre das definitiv erstrebenswert, um keine Middleware betreiben zu müssen.

**Comment by hesne** [comment]
damals wurde das Ganze aber auch noch nicht mit SPRING gemacht, oder?

**Comment by zlex650** [comment]
Soweit ich weiß, schon. Der aufbau der Parameter ist ähnlich wie bei der Urlaubsverwaltung. Die Probleme liegen hier an der Scope Config, da die Rechte über diese Zuordnung gesetzt werden. anders als bei der Urlaubsverwaltung. Diese ist so nicht abbildbar in Azure, zumindest habe ich das nicht hinbekommen.

Ich habe seit damals die gleiche Konfig und fahre die neuste Version. Also SPRING war schon immer die Basis.

**Reply by hesne** [reply]
Ach bei der Urlaubsverwaltung klappt es direkt mit Azure?

**Comment by zlex650** [comment]
Ja, da ist das kein Problem

**Reply by hesne** [reply]
Gibt es hierzu eine Anleitung? Ich hatte es ad hoc nicht zum laufen bekommen....

**Reply by zlex650** [reply]
Gibt es hier ein Möglichkeit seine Kontaktdaten zu teilen um direkt zu kommunizieren ohne das diese öffentlich sind? Würde dir dann ein paar wichtige Infos zukommen lassen.

**Reply by grafjo** [reply]
@hesne magst du dazu bei der Urlaubsverwaltung eine Diskussion aufmachen?

**Comment by hesne** [comment]
Ist schon gelöst.
________________________________
Von: grafjo ***@***.***>
Gesendet: Thursday, November 21, 2024 9:24:12 PM
An: urlaubsverwaltung/zeiterfassung ***@***.***>
Cc: Henning Eßer ***@***.***>; Mention ***@***.***>
Betreff: Re: [urlaubsverwaltung/zeiterfassung] OIDC mit Azure AD Auth ohne Keycloak? (Discussion #348)


@hesne<https://github.com/hesne> magst du dazu bei der Urlaubsverwaltung eine Diskussion aufmachen?

**Comment by grafjo** [comment, marked as answer]
@zlex650 @hesne mit https://github.com/urlaubsverwaltung/zeiterfassung/releases/tag/zeiterfassung-2.22.0 unterstützen wir nun u.a. auch entra id
